.travis.yml

```diff
@@ -29,3 +29,4 @@ script:
   - export MXNET_STORAGE_FALLBACK_LOG_VERBOSE=0
   - mv make/osx.mk config.mk
   - make -j 2
```

Comment:
@KellenSunderland instead of placing python here as a job that travis would always execute, could you please put python build section into a separate target so we can also add Scala-maven test in here in the future.

Reply:
Possibly this will help: https://docs.travis-ci.com/user/build-stages

Reply:
I think it would be great if we could add Scala tests as well, but I do have a few concerns.  How long would they take to execute?  Let's keep in mind that we have a limited number of executors that can be run before we begin interfering with other Apache projects.  There's also a global timeout of ~45 minutes to build and execute on a virtualized MacOS host which takes a few minutes just to start.  I'd ideally like to keep our total test time below 30 minutes to avoid hitting this in the future.

In principle I'm not against build stages, but unless we have a lot of jobs running in parallel that have dependencies on each other I don't see a big benefit in using them.

Reply:
Thanks for your reply! If you mean `scalaunittest`it will be around 10 mins or less. Is (travis wait)[https://stackoverflow.com/questions/43918874/how-to-increase-no-activity-wait-time-in-travis-ci]  possible to apply in here?

Reply:
Unfortunately Travis wait won't help in this case.  The tests are active and sending output to Travis, they just take a long time and hit the global timeout for Travis jobs.

10 minutes actually is reasonable.  Let's do a test run with Scala enabled after this is merged and see if we can get it included.  My guess is if we enable folder caching, and cache models we're currently downloading we'll be able to fit it in and not hit the global timeouts.